raiden/network/transport/matrix/utils.py

```diff
@@ -294,18 +312,25 @@ def _maybe_address_reachability_changed(self, address: Address) -> None:
 
         new_address_reachability = USER_PRESENCE_TO_ADDRESS_REACHABILITY[new_presence]
 
-        prev_addresss_reachability = self.get_address_reachability(address)
-        if new_address_reachability == prev_addresss_reachability:
+        prev_reachability_state = self.get_address_reachability_state(address)
+        if new_address_reachability == prev_reachability_state:
```

Comment:
This is a bug. It's comparing an `AddressReachability` to an `AddressReachabilityState`.

Reply:
Why didn't mypy find this?

Reply:
Yeah I was wondering about that as well. 
Apparently because we [haven't enabled `--strict-equality`](https://mypy.readthedocs.io/en/latest/command_line.html#cmdoption-mypy-strict-equality)

Reply:
Damn man ... some times the mypy defaults make no sense. I also have seen things that mypy should have caught but for some reason did not.